doc/dev/logging.md

```diff
@@ -0,0 +1,96 @@
+
+In the above example, `logf.ZapLogger()` takes a boolean flag to set development parameters. Passing in `true` will set the logger to log in development mode; debug log statements will trigger, and error log statements will include stack traces.
+
+## Creating a structured log statement
+
+There are two ways to create structured logs with `logr`. You can create new loggers using `log.WithValues(keyValues)` that include `keyValues`, a list of key-value pair `interface{}`'s, in each log record. Alternatively you can include `keyValues` directly in a log statement, as all `logr` log statements take some message and `keyValues`. The signature of `logr.Error()` has an `error`-type parameter, which can be `nil`.
+
+An example from [`memcached_controller.go`][code_memcached_controller]:
+
+```Go
+package memcached
+
+// Set a global logger for the memcached package. Each log record produced
+// by this logger will have an identifier containing "controller_memcached".
+// These names are hierarchical; the name attached to memcached log statements
+// will be "operator-sdk.controller_memcached" because SDKLog has name
+// "operator-sdk".
+var log = logutil.SDKLog.WithName("controller_memcached")
+
+func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+  // Create a logger for Reconcile() that includes "Request.Namespace"
+  // and "Request.Name" in each log record from this log statement.
+  reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+  reqLogger.Info("Reconciling Memcached.")
+
+  memcached := &cachev1alpha1.Memcached{}
+  err := r.client.Get(context.TODO(), request.NamespacedName, memcached)
+  if err != nil {
+    if errors.IsNotFound(err) {
+      reqLogger.Info("Memcached resource not found. Ignoring since object must be deleted.")
+      return reconcile.Result{}, nil
+    }
+    return reconcile.Result{}, err
+  }
+
+  found := &appsv1.Deployment{}
+  err = r.client.Get(context.TODO(), types.NamespacedName{Name: memcached.Name, Namespace: memcached.Namespace}, found)
+  if err != nil {
+    if errors.IsNotFound(err) {
+      dep := r.deploymentForMemcached(memcached)
+      // Include "Deployment.Namespace" and "Deployment.Name" in records
+      // produced by this particular log statement. "Request.Namespace" and
+      // "Request.Name" will also be included from reqLogger.
+      reqLogger.Info("Creating a new Deployment", "Deployment.Namespace", dep.Namespace, "Deployment.Name", dep.Name)
+      err = r.client.Create(context.TODO(), dep)
+      if err != nil {
+        // Include the error in records produced by this log statement.
+        reqLogger.Error(err, "failed to create new Deployment", "Deployment.Namespace", dep.Namespace, "Deployment.Name", dep.Name)
+        return reconcile.Result{}, err
+      }
+    }
+    return reconcile.Result{}, err
+  }
+
+  ...
+}
+```
```

Comment:
@estroz 👍 The doc looks great.
Can we also add a sample output of the logs at the end of this snippet to show what the info and error logs look like with the keyValues set. We can mention that it's for the default zap implementation.